ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -3580,7 +3592,19 @@ public boolean dropPartition(String dbName, String tableName, List<String> parti
     List<String> pvals = MetaStoreUtils.getPvals(t.getPartCols(), partSpec);
 
     try {
-      names = getMSC().listPartitionNames(dbName, tblName, pvals, max);
+      if (AcidUtils.isTransactionalTable(t)) {
+        GetPartitionNamesPsRequest req = new GetPartitionNamesPsRequest();
+        req.setTblName(tblName);
+        req.setDbName(dbName);
+        req.setPartValues(pvals);
+        req.setMaxParts(max);
+        ValidWriteIdList validWriteIdList = getValidWriteIdList(dbName, tblName);
+        req.setValidWriteIdList(validWriteIdList != null ? validWriteIdList.toString() : null);
+        GetPartitionNamesPsResponse res = getMSC().listPartitionNamesRequest(req);
+        names = res.getNames();
+      } else {
+        names = getMSC().listPartitionNames(dbName, tblName, pvals, max);
```

Comment:
Maybe use the same listPartitionNamesReques for acid and non-acid tables as well?

Reply:
Makes sense, done.